core/src/main/java/ai/timefold/solver/core/impl/domain/solution/cloner/PlanningCloneable.java

```diff
@@ -0,0 +1,5 @@
+package ai.timefold.solver.core.impl.domain.solution.cloner;
+
+public interface PlanningCloneable<T> {
+    T createEmptyInstance();
```

Comment:
I'd like this interface to have wider use. It should not only serve for collections, it should serve for any type. 

To that end, the method should be called `createNewInstance()`, and the cloner should be adapted to handle it. As a test case, if it can clone a cloneable planning entity inside of a non-cloneable planning solution, that will be enough proof.